op-acceptance-tests/tests/interop/loadtest/executor.go

```diff
@@ -0,0 +1,186 @@
+package loadtest
+
+import (
+	"math/big"
+	"slices"
+	"sync"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/devnet-sdk/contracts/constants"
+	"github.com/ethereum-optimism/optimism/op-devstack/devtest"
+	"github.com/ethereum-optimism/optimism/op-devstack/dsl"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/txintent"
+	"github.com/ethereum-optimism/optimism/op-service/txplan"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+)
+
+type Executor interface {
+	Execute(t devtest.T, msgs []types.Message)
+}
+
+type ValidExecutor struct {
+	el  *dsl.L2ELNode
+	eoa *dsl.EOA
+	// supervisor is used to check if executing messages are cross-safe.
+	supervisor *dsl.Supervisor
+
+	counter *counter
+}
+
+func NewValidExecutor(funder *dsl.Funder, el *dsl.L2ELNode, supervisor *dsl.Supervisor) *ValidExecutor {
+	return &ValidExecutor{
+		el:         el,
+		eoa:        funder.NewFundedEOA(eth.MillionEther),
+		supervisor: supervisor,
+		counter:    new(counter),
+	}
+}
+
+func (e *ValidExecutor) Execute(t devtest.T, msgs []types.Message) {
+	tx := buildExecTx(e.eoa, e.el, msgs, retryForever(e.el.Escape().EthClient()), txplan.WithStaticNonce(e.counter.Next()))
+	receipt, err := tx.Included.Eval(t.Ctx())
+	t.Require().NoError(err)
+	_, err = tx.Success.Eval(t.Ctx())
+	t.Require().NoError(err)
+	t.Require().Len(receipt.Logs, len(msgs))
+
+	// Wait for the transaction to be cross-safe.
+	includedBlock, err := tx.IncludedBlock.Eval(t.Ctx())
+	t.Require().NoError(err)
+	for {
+		// NOTE: it may be desirable to query proxyd instead of the supervisor if/when the devstack supports it.
+		crossSafeID, err := e.supervisor.Escape().QueryAPI().CrossSafe(t.Ctx(), e.el.ChainID())
+		t.Require().NoError(err)
+		if includedBlock.ID().Number <= crossSafeID.Derived.Number {
+			break
+		}
+		e.el.WaitForBlock()
+	}
```

Comment:
Would be better if waiting were decoupled from transaction relaying, so that we might relay more than len(executors) transactions' worth of messages at once.

Reply:
Relaying is unbounded since it's done in parallel (unless I'm missing something).

Separating this could still make sense to take pressure off the supervisor RPC, if that's desired. The ideal solution is to ping proxyd instead of the actual supervisor.